ALU_wrapper.py

```diff
@@ -82,47 +63,36 @@ def sendtoslims():
+    destination = seqpath + "/saved/"  # This needs changing. Now set to sub-folder saved under input-path (seqpath).
+    filename = folderpath.split('/').pop() + ".tar.gz"
     subprocess.call(['tar', 'zcf', destination + filename, folderpath])
 
 
-if not args.run:
-    print("Not parsing any found files, specify -r to actually run")
-folderList = next(os.walk(seqpath + '/.'))[1]
-if args.verbose:
-    print("List of folders in dir: ")
-    for folder in folderList:
-        print(folder)
-    print("--- END OF LIST ---")
+def mailout():
+    msg_from = "crc@gu.se"  # TODO: Get from config-file
+    msg_subject = "The crc files are done!"
+    msg_body = "Hi!\nThe CLC-files are processed and ready for download."  # TODO: Change to a better mail
+    emailer.send_email(msg_to, msg_from, msg_subject, msg_body)
 
+
+folderList = next(os.walk(seqpath + '/.'))[1]
 for folder in folderList:
+    logging.info("Folder: " + folder)
     if "DNA" in folder and "_" not in folder:
-        # Skipping folders with _ in the names... this is actually not needed.
+        # Skipping folders with _ in the names. This is not actually needed.
         path = seqpath + '/' + folder
-        if glob.glob(path + '/*melt*.vcf*'):  # depending on how we want to run this. just an example.
-            if args.verbose:
-                print("Processed files found in folder " + folder + ", not running again.")
+        if glob.glob(path + '/*melt*.vcf*'):  # TODO: Get regex from config-file
+            logging.debug("Processed files found in folder " + folder + ", not running again.")
         else:
-            if args.verbose:
-                print("No processed files found in folder " + folder + ", continuing.")
+            logging.debug("No processed files found in folder " + folder + ", continuing.")
             if args.run:
-                if args.verbose:
-                    print("running on folder: " + path)
-                print("Folder: " + folder + " Path: " + path)
+                logging.debug("running on folder: " + folder + "; path: " + path)
                 alurunner()
                 sendtoslims()
                 compressfiles(path)
-                sendmail = True
+                finished = True
             else:
-                if args.verbose:
-                    print("Not running on folder " + path + ", use -r to actually run")
-
-if sendmail:
-    mail_from = ""
-    mail_subject = "the crc files are done"
-    mail_body = "Hi!\nThe CLC-files are processed and ready for download."  # Change to a better mail
-    #  email_general(msg_to, msg_from, subject, body) # Uncomment to actually send mail.
-    if args.verbose:
-        print("mailto: " + mail_to + "; mail from: " + mail_from +
-              "; subject: " + mail_subject + ";\n--- mail body ---\n" + mail_body)
+                logging.debug("Not running on folder " + path + ", use -r to actually run")
+
+if finished:
+    mailout()
```

Comment:
To me this logic is a bit roundabout.
The mailout function contains all logic for find all the folders, calling all the other functions etc. 
Wouldn't it feel much cleaner to have discreet steps for finding folders, running the alirunn, send to slims etc in a mina function.

And in the end of this function, if there has been any new data process, then call a mail function.

That would make the process much easier to follow.

Reply:
Will review this function in a later PR